src/main/java/com/puppycrawl/tools/checkstyle/xpath/ElementNode.java

```diff
@@ -82,10 +82,8 @@ public ElementNode(AbstractNode root, AbstractNode parent, DetailAST detailAst)
     private void createChildren() {
         DetailAST currentChild = detailAst.getFirstChild();
         while (currentChild != null) {
-            if (currentChild.getType() != TokenTypes.IDENT) {
-                final ElementNode child = new ElementNode(root, this, currentChild);
-                addChild(child);
-            }
+            final ElementNode child = new ElementNode(root, this, currentChild);
```

Comment:
Can we use AbstractNode instead of ElementNode? We need to use iterfaces or abstract classes where possible.

Reply:
Done